### PR TITLE
Fix DB and Connected R repo

### LINKED ISSUE #63: Connect the backend with the R repository
This issue is **BLOCKED** by https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-3/issues/62
Try to connect the backend to use the R directory

### PR DESCRIPTION
**Pull request by ralphgregorio**
Resolves #63 

**TLDR: This PR makes our backend fully functional without errors! yay!**

Summary of changes in this pr:
- Fixed the schema in the database by properly creating functions and tables with its constraints
- Installed pgcrypto
- Connected the R repository to our backend
- Added installation of R packages to our backend container
- Added initial data needed for our tables
- Fixed some errors in the java objects

With these changes `docker-compose build` will now take some time to fully build on first build. Depending on your computer its estimated to take around 5 - 20 minutes to fully build the containers in your local environment.

I did notice that the java backend is buggy and would need some looking into to properly fix it. For now I put some placeholder as a bandaid fix but we will need a proper fix soon. I will create an issue regarding this.

I tested that the api fully works by using the old BeapEngineUI with our backend. And The login, uploading, processing and predicting routes should now fully work.

If you want to test this on your local machine you will have to rebuild your containers by `docker-compose up --build`

### CODE REVIEW

**Review by ArdalanAskarian — APPROVED** [review]
Everything is well documented and works as expected.
